lib/pdk/util/puppet_version.rb

```diff
@@ -59,6 +59,36 @@ def from_pe_version(version_str)
         find_gem_for(gem_version[:gem_version])
       end
 
+      def from_module_metadata(metadata)
```

Comment:
shouldn't this parsing be located in the `PDK::Module::Metadata` class?

Reply:
It could go in either really, but I opted to contain it with all the other puppet version finding logic for clarity (imho). I can move it to be a method of `PDK::Module::Metadata` if that's the group consensus though.

Reply:
I've moved the validation logic into `PDK::Module::Metadata` and left the version finding logic in `PDK::Util::PuppetVersion`